packages/nimble-components/src/spinner/specs/README.md

```diff
@@ -0,0 +1,138 @@
+# Nimble Spinner
+
+## Overview
+
+Spinner component (`nimble-spinner`): Indeterminate progress indicator / loading indicator. Meant to be temporarily used in a region of a web page until that region's contents have loaded, or to represent an ongoing operation that doesn't have a known/ determinate duration.
+
+### Background
+
+[Nimble spinner: #346](https://github.com/ni/nimble/issues/346)  
+[Nimble spinner design: #822](https://github.com/ni/nimble/issues/822)  
+[Visual Design spec - Adobe XD](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/dece308f-79e7-48ec-ab41-011f3376b49b/)
+
+### Non-goals
+
+Ability to show determinate progress:
+
+-   The current visual design for the spinner is distinct/different from an indeterminate radial progress bar
+-   We don't currently have visual designs for a linear or radial progress bar with determinate progress
+-   If/when we want progress bar components in the future, at that point we'll probably create new Nimble components which extend the corresponding FAST components:
+    -   `nimble-progress` or `nimble-linear-progress`, deriving from `fast-progress`, for a linear progress bar; and
+    -   `nimble-progress-ring` or `nimble-radial-progress`, deriving from `fast-progress-ring`, for a radial progress bar
+
+### Features
+
+-   Show indeterminate progress with an animation
+    -   Cannot be paused - assumption is that component will be added to/ removed from the page as needed
+-   Supports 3 sizes: `small` (16x16), `medium` (32x32) (default), `large` (64x64)
+-   For the color theme only, supports an alternate appearance/color scheme via `appearance-variant="alternate"`
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+SystemLink Enterprise uses a spinner based on the Material `mat-spinner`:  
+![SystemLink Material spinner](SlMatSpinner.gif)
+
+A variant of this spinner design is already implemented in the Web Config tool written in Blazor:  
+![Blazor Web Config spinner](BlazorWebConfigSpinner.gif)
+
+## Design
+
+The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent loading progress, or an ongoing operation, of an indeterminate / unknown duration.
+
+### API
+
+Component Name: `nimble-spinner`
+
+Properties/Attributes:
+
+-   `size`
+    -   String attribute controlling size of the component
+    -   Backed by an enum `SpinnerSize` with values `small` (16x16), `medium` (32x32), `large` (64x64)
+    -   If omitted, the default is `medium` (32x32)
```

Comment:
We should provide "presets" to enable easy consistency. Ideally, we also provide a way to override that so something custom could be used. Atlassian shows a good approach for this (along with use case recommendation for each size)

https://atlassian.design/components/spinner/examples#sizes

Reply:
@rajsite I feel like the sizes are somewhat arbitrary, maybe we can discuss with @NIbokeefe tomorrow.  
As currently implemented, it should be pretty easy to support arbitrary sizes, though if the width/height aren't a multiple of 4 we'd end up with floating point numbers for translation offsets and such.

Have we decided on a pattern for a client-configured component size in the cases where we need to refer to it with CSS variables for computations? (The drawer's CSS variable is still marked 'not recommended to override' but don't remember if we came up with a different approach)

Reply:
No our sizing is currently kinda broken. This issue captures it: https://github.com/ni/nimble/issues/610

The summary is that the direction I think we should go is using tokens for the default value of sizes (for consistency across controls), attribute enums for control-specific alternatives, and then normal css for customization.

In very pseduocode that won't work in practice because of labels and things but conceptually:
```css
: host {
   height: var(--ni-nimble-control-height);
}
:host([size="small"]) {
    height: 100px;
}
```

So in the end we don't have a proprietary API based on setting custom properties to configure control size but instead good defaults and then you can override with normal css

```html
<my-element> <!-- default size -->
<my-element size="small"> <!-- choose from preset -->
<my-element style="width: 128px; height: 128px;"> <!-- customize the size -->
```

In practice we probably need to expose and document `css parts` that you can style but the main difference is styling those parts with css vs custom properties for customization.

Reply:
I was able to update my spinner CSS/animation to handle custom sizes without a problem, so we can discuss tomorrow.
Added sizing as an Open Issue and added your proposal to the spec (which is basically making 'size' optional, in which case user can specify width/height)

Reply:
Following up on sizing discussion from earlier. @NIbokeefe pointed out that the bit sizes don't take up the full 16x16/32x32/64x64, and there's some margin/padding that he chose specifically (varies a bit at each size).

I implemented that sizing behavior locally. I think we could still give the component a reasonable layout at user-provided sizes. 32x32 and 64x64 follow the same pattern (6.25% margins, 87.5% for the bounding box of the bits). Smaller sizes are a little different though / have more margin (that formula yields 1px per side but XD has 2px instead).
So I'm still fine with requiring `size` (allowing only the 3 sizes specified), but giving the option to override still seems potentially OK too (we just ensure we use the designer-provided sizes and margins at the 3 preset sizes). @rajsite @jattasNI any thoughts?

Reply:
As long as the computed Brandon thinks the computed margins look good at the sizes we think are likely to be used, I think that's a great approach.

Reply:
Updated spec to say that we'll support these 3 sizes, but the component can scale to other sizes (advanced use case + consumer should doublecheck padding with Design).